openmp/libomptarget/include/PluginManager.h

```diff
@@ -208,6 +213,13 @@ struct PluginManager {
 
   /// Devices associated with plugins, accesses to the container are exclusive.
   ProtectedObj<DeviceContainerTy> Devices;
+
+  /// Whe running on an APU, the GPU plugin may decide to
+  /// run in zero-copy even though the user did not program
+  /// their application with unified_shared_memory requirement.
+  /// This execution modality is called auto zero-copy and it is
+  /// cached information from the Plugin.
+  bool UseAutoZeroCopy = false;
```

Comment:
I'd figure this is a device property. I can easily imagine an APU system with a dedicated card and this property should not apply to all devices. If we make it global, it doesn't need a flag, we could simply implicitly set OMP_REQ_UNIFIED_SHARED_MEMORY in the requirements, no?

Reply:
Fair enough: I do not have such a card to test against, but it is conceivable that someone will eventually have it. What I am thinking is a check on every device in the system to see if they are all an APU and would turn auto zero-copy on, then we turn it on in libomptarget. If one or more of the devices is not an APU, then it will return false to the query and we will not do auto zero-copy.
I think this should cover any future arch and we can think about mixing and matching based on the device num that is passed to getTargetPointer and getTgtPtr in a later patch.

Reply:
I'm fine with doing it per plugin for now, but I would like to avoid the extra bool here and the getter/setter. Can we add a new requirements flag:
`OMPX_REQ_AUTO_ZERO_COPY`
and reuse the requirement stuff instead of adding a on-off bool in many places?

Reply:
I am happy to have a new OMP_REQ_AUTO_ZERO_COPY requirement flag.
I am already done moving the property to a device one and I would have just to change this to set the req flag instead of having the boolean check. Will do that today if this makes sense.